src/Type/Product.php

```diff
@@ -78,19 +82,33 @@ private function getCurrentProduct(): ?ProductInterface
      */
     private function getStoreProduct(
         int $storeId
-    ): ProductModel {
+    ): ?ProductInterface {
         $currentStore   = $this->storeManager->getStore();
         $currentProduct = $this->getCurrentProduct();
+        $websiteId      = $this->getWebsiteByStoreId($storeId);
+
+        if (!in_array($websiteId, $currentProduct->getWebsiteIds() ?? [])) {
+            return null;
```

Comment:
You can reduce the return points and make it the last code point in the method, by putting the return value in a variable.

```php
$result = null;

if (in_array($websiteId, $currentProduct->getWebsiteIds() ?? [], true)) {
    try {
        $result = ...
    } catch (NoSuchEntityException $e) {
        $result = null;
    }
}

return $result;
```